crates/typed-store/src/rocks/mod.rs

```diff
@@ -1825,14 +1864,40 @@ pub fn base_db_options() -> DBOptions {
     }
 
     // The table cache is locked for updates and this determines the number
-    // of shards, ie 2^6. Increase in case of lock contentions.
+    // of shards, ie 2^10. Increase in case of lock contentions.
     opt.set_table_cache_num_shard_bits(10);
 
+    // Allow buffering 1GiB of writes before flushing to disk.
+    opt.set_write_buffer_size(256 * 1024 * 1024);
+    opt.set_max_write_buffer_number(4);
+
+    opt.set_universal_compaction_options(&UniversalCompactOptions::default());
+    // Compaction trigger for universal compaction. Default to 8.
+    opt.set_level_zero_file_num_compaction_trigger(
+        read_size_from_env(ENV_VAR_L0_NUM_FILES_COMPACTION_TRIGGER)
+            .unwrap_or(DEFAULT_L0_NUM_FILES_COMPACTION_TRIGGER)
+            .try_into()
+            .unwrap(),
+    );
+    opt.set_num_levels(8);
+    // ~512MiB per sst file.
+    opt.set_target_file_size_base(512 * 1024 * 1024);
```

Comment:
I wonder what the best way is to properly test this, since you may see all the benefits now but discover potential downsides much later. 
With the increasing size of SST files and the use of level-style compaction, compactions will become less frequent. However, they will still occur and when they do, they will be slower (more data to merge), require more memory usage, and likely causing write stalls. 
Also space reclamation will become even more problematic than it is today

Reply:
I think sst compaction memory usage should not change much with the size of sst files. I raised the level 0 write stall thresholds, and lmk if there are other levers to control when write stalls happen. I hope with larger write buffers, in-epoch space reclamation can happen more often in memory. For cross epoch space reclamation on perpetual tables, I'm not sure how sst size can matter since the node still needs to compact across all data from an epoch.

For testing, maybe the best way would be running the network with high write load for longer. I can decrease the default to 256MB for now and make this into an env var as well.

Reply:
> compaction memory usage should not change much with the size of sst files

I think it does and it's getting worse on deeper levels. The problem is getting initially delayed though. 

Not sure what's the best method for picking up the value. Just maybe for reference, NEAR set it to 64, Solana to ~100. 
Making it configurable via env var is a great idea.

And btw looks like we stuck with cross-epoch reclamations for some time

Reply:
My mental model is that merging N sst files has memory usage of N * prefetch size. I guess it can be different if there are more metadata that needs to be tracked. I will lower this to 128MiB for now and check again in throughput experiments.